manager/src/main/java/com/redhat/service/bridge/manager/vault/AWSVaultServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+
+import io.vertx.core.json.Json;
+
+@ApplicationScoped
+public class AWSVaultServiceImpl implements VaultService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AWSVaultServiceImpl.class);
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.region")
+    String region;
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.endpoint-override")
+    Optional<String> endpoint;
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.access-key-id")
+    String accessKeyId;
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.secret-access-key")
+    String secretAccessKey;
+
+    private AWSSecretsManager awsSecretsManager;
+    private SecretCache secretCache;
+
+    @PostConstruct
+    void init() {
+        AWSCredentialsProvider credentials = new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(accessKeyId, secretAccessKey));
+        AWSSecretsManagerClientBuilder clientBuilder = AWSSecretsManagerClientBuilder
+                .standard()
+                .withCredentials(credentials);
+        if (endpoint.isPresent()) {
+            clientBuilder.withEndpointConfiguration(
+                    new AwsClientBuilder.EndpointConfiguration(endpoint.get(), region));
+        } else {
+            clientBuilder.withRegion(region);
+        }
+
+        AWSSecretsManager client = clientBuilder.build();
+        SecretCacheConfiguration cacheConf = new SecretCacheConfiguration()
+                .withMaxCacheSize(SecretCacheConfiguration.DEFAULT_MAX_CACHE_SIZE)
+                .withCacheItemTTL(SecretCacheConfiguration.DEFAULT_CACHE_ITEM_TTL)
+                .withClient(client);
+        this.awsSecretsManager = client;
+        this.secretCache = new SecretCache(cacheConf);
+    }
+
+    @Override
+    public void createOrReplace(EventBridgeSecret secret) {
+        try {
+            CreateSecretRequest createSecretRequest = new CreateSecretRequest()
+                    .withName(secret.getId())
+                    .withSecretString(Json.encode(secret.getValues()));
+            awsSecretsManager.createSecret(createSecretRequest);
+
+            LOGGER.debug("Secret '{}' created in AWS Vault", secret.getId());
+        } catch (ResourceExistsException e) {
+            LOGGER.info("Secret '{}' already exists in AWS Vault. Replacing..", secret.getId());
+            PutSecretValueRequest putSecretValueRequest = new PutSecretValueRequest()
+                    .withSecretId(secret.getId())
+                    .withSecretString(Json.encode(secret.getValues()));
+            awsSecretsManager.putSecretValue(putSecretValueRequest);
+            LOGGER.info("Secret '{}' already exists in AWS Vault. Replaced", secret.getId());
+        } catch (InvalidRequestException e) {
+            String message = String.format("Unable to create secret '%s' in AWS Vault", secret.getId());
+            LOGGER.error(message, e);
```

Comment:
Let's make our centralised error handling log the Error. That means we can just throw the `Exception` in the code here